objects/src/builder/mod.rs

```diff
@@ -0,0 +1,274 @@
+}
+
+impl<T: Rng> NonFungibleAssetBuilder<T> {
+    pub fn new(faucet_id: AccountId, rng: T) -> Result<Self, AssetError> {
+        let details_builder = NonFungibleAssetDetailsBuilder::new(faucet_id, rng)?;
+        Ok(Self { details_builder })
+    }
+
+    pub fn with_details(
+        &mut self,
+        details: &NonFungibleAssetDetails,
+    ) -> Result<NonFungibleAsset, AssetError> {
+        NonFungibleAsset::new(&details)
+    }
+
+    pub fn build(&mut self) -> Result<NonFungibleAsset, AssetError> {
+        let details = self.details_builder.build()?;
+        NonFungibleAsset::new(&details)
+    }
+}
+
+/// Builder for an `Account`, the builder allows for a fluent API to construct an account. Each
+/// account needs a unique builder.
+pub struct AccountBuilder<T: Rng> {
+    assets: Vec<Asset>,
+    storage_builder: AccountStorageBuider,
+    code: String,
+    nonce: Felt,
+    account_id_builder: AccountIdBuilder<T>,
+}
+
+const DEFAULT_ACCOUNT_CODE: &str = "\
+use.miden::sat::account
+
+export.incr_nonce
+    push.0 swap
+    # => [value, 0]
+
+    exec.account::incr_nonce
+    # => [0]
+end
+
+export.set_item
+    exec.account::set_item
+    # => [R', V, 0, 0, 0]
+
+    movup.8 drop movup.8 drop movup.8 drop
+    # => [R', V]
+end
+
+export.set_code
+    padw swapw
+    # => [CODE_ROOT, 0, 0, 0, 0]
+
+    exec.account::set_code
+    # => [0, 0, 0, 0]
+end
+
+export.account_procedure_1
+    push.1.2
+    add
+end
+
+export.account_procedure_2
+    push.2.1
+    sub
+end
+";
+
+impl<T: Rng> AccountBuilder<T> {
+    pub fn new(rng: T) -> Self {
+        Self {
+            assets: vec![],
+            storage_builder: AccountStorageBuider::new(),
+            code: DEFAULT_ACCOUNT_CODE.to_string(),
+            nonce: ZERO,
+            account_id_builder: AccountIdBuilder::new(rng),
+        }
+    }
+
+    pub fn add_asset(mut self, asset: Asset) -> Self {
+        self.assets.push(asset);
+        self
+    }
+
+    pub fn add_storage_item(mut self, item: StorageItem) -> Self {
+        self.storage_builder.add_item(item);
+        self
+    }
+
+    pub fn code<C: AsRef<str>>(mut self, code: C) -> Self {
+        self.code = code.as_ref().to_string();
+        self
+    }
+
+    pub fn nonce(mut self, nonce: Felt) -> Self {
+        self.nonce = nonce;
+        self
+    }
+
+    pub fn account_type(&mut self, account_type: AccountType) -> &mut Self {
+        self.account_id_builder.account_type(account_type);
+        self
+    }
+
+    pub fn on_chain(&mut self, on_chain: bool) -> &mut Self {
+        self.account_id_builder.on_chain(on_chain);
+        self
+    }
+
+    pub fn code_root(&mut self, code_root: Digest) -> &mut Self {
+        self.account_id_builder.code_root(code_root);
+        self
+    }
+
+    pub fn storage_root(&mut self, storage_root: Digest) -> &mut Self {
+        self.account_id_builder.storage_root(storage_root);
+        self
+    }
+
+    pub fn build(mut self) -> Result<Account, AccountError> {
```

Comment:
As with my comments on `AccountIdBuilder`, for new accounts we will need to return the account seed to the client so we will need to either redefine this function signature or introduce a new builder method for new accounts.

Reply:
same comment: https://github.com/0xPolygonMiden/miden-base/pull/206#discussion_r1305314343